receiver/statsdreceiver/transport/transport.go

```diff
@@ -43,3 +44,12 @@ func (trans Transport) IsPacketTransport() bool {
 	}
 	return false
 }
+
+// Returns true if the transport is unix socket based.
+func (trans Transport) IsUnixTransport() bool {
+	switch trans {
```

Comment:
NIT isn't the switch an overkill?

Reply:
You are missing protocols that can opo like `unix`, `unixgram`... I kept the same structure for all functions.